treeherder/intermittents_commenter/commenter.py

```diff
@@ -0,0 +1,285 @@
+        with open('treeherder/intermittents_commenter/{}'.format(filename), 'r') as myfile:
+            if load:
+                return json.load(myfile)
+            else:
+                return myfile.read()
+
+    def calculate_date_strings(self, mode, numDays):
+        """Returns a tuple of start (in YYYY-MM-DD format) and end date
+           strings (in YYYY-MM-DD HH:MM:SS format for an inclusive day)."""
+
+        yesterday = date.today() - timedelta(days=1)
+        endday = datetime(yesterday.year, yesterday.month, yesterday.day, 23, 59, 59, 999)
+        if mode:
+            startday = yesterday - timedelta(days=numDays)
+        else:
+            # daily mode
+            startday = yesterday
+
+        return startday.isoformat(), endday.strftime('%Y-%m-%d %H:%M:%S.%f')
+
+    def check_whiteboard_status(self, whiteboard):
+        """Extracts stockwell text from a bug's whiteboard status to
+           determine whether it matches specified stockwell text;
+           returns a boolean"""
+
+        stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
+        if stockwell_text is not None:
+            text = stockwell_text.group(1).split(':')[0]
+            if text == 'fixed' or text == 'disable-recommended' or text == 'infra' or text == 'disabled':
+                return True
+        return False
+
+    def update_whiteboard(self, existing, new):
+        return re.sub('\[stockwell.*?\]', new, existing)
+
+    def new_request(self):
+        session = requests.Session()
+        # Use a custom HTTP adapter, so we can set a non-zero max_retries value.
+        session.mount("https://", requests.adapters.HTTPAdapter(max_retries=3))
+        session.headers = {
+            'User-Agent': TREEHERDER_USER_AGENT,
+            'x-bugzilla-api-key': settings.COMMENTER_API_KEY,
+            'Accept': 'application/json'
+        }
+        return session
+
+    def create_url(self, bug_id):
+        return settings.BZ_API_URL + '/rest/bug/' + str(bug_id)
+
+    def fetch_bug_details(self, params, bug_id):
+        """fetches bug metadata from bugzilla and returns an encoded
+           dict if successful, otherwise returns None"""
+
+        try:
+            response = self.session.get(self.create_url(bug_id), headers=self.session.headers, params=params,
+                                        timeout=REQUESTS_TIMEOUT)
+            response.raise_for_status()
+        except RequestException as e:
+            logger.warning('error fetching bugzilla metadata for bug {} because {}'.format(bug_id, e))
+            return None
+
+        # slow down: bmo server may refuse service if too many requests made too frequently
+        time.sleep(0.5)
+        data = response.json()
+        if 'bugs' not in data:
+            return None
+
+        return {key.encode('UTF8'): value.encode('UTF8') for key, value in data['bugs'][0].iteritems()}
+
+    def submit_bug_comment(self, params, bug_id):
```

Comment:
Somewhere in here we need to handle `settings.COMMENTER_API_KEY` being `None`, so that we don't get errors on stage/dev, where we intentionally won't set the API key, to prevent duplicate bug comments.

(Alternatively, this check could be added to the caller of `submit_bug_comments`, alongside the check for `dry_run` -- though unlike for dry run, we probably don't want to spam a load of print statements, or we'll fill up Papertrail with a lot of noise)